src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/RazorDefinitionEndpoint.cs

```diff
@@ -165,29 +237,49 @@ public DefinitionRegistrationOptions GetRegistrationOptions(DefinitionCapability
             if (node is null)
             {
                 logger.LogInformation("Could not locate ancestor of type MarkupTagHelperStartTag or MarkupTagHelperEndTag.");
-                return null;
+                return (null, null);
             }
 
             var name = GetStartOrEndTagName(node);
             if (name is null)
             {
                 logger.LogInformation("Could not retrieve name of start or end tag.");
-                return null;
+                return (null, null);
+            }
+
+            string? propertyName = null;
+
+            // If we're on an attribute then just validate against the attribute name
+            if (owner.Parent is MarkupTagHelperAttributeSyntax attribute)
```

Comment:
We should also cross-check [the minimized variant](https://github.com/dotnet/razor-compiler/blob/07e115540fd82508bf261ea85adc55734b12cdb9/src/Microsoft.AspNetCore.Razor.Language/src/Syntax/Syntax.xml#L199). Lol I really wish we just had a bool or something on the base TagHelperAttribute syntax 😢

Reply:
Yeah, minimized versions of things are a pain. This also raises the question of what about MarkupTagHelperDirectiveAttribute? Certainly a less common scenario, but I believe it's still possible to have in user-defined code.

Reply:
> Certainly a less common scenario, but I believe it's still possible to have in user-defined code.

Ya I'm actually not sure what the expectations are there. At the very least we can leave it as a follow up but I'm curious as well. @TanayParikh are folks able to define their own directive attributes?

Reply:
Well TIL that minimized attributes exist, and something about dictionaries?! To google!

(But not now, I'm still in bed)